### ISSUE TITLE
Empty filters no longer allowed, breaking our pipelines 

### ISSUE THREAD

**Issue by jakobatfreiheit** [hashicorp/terraform-provider-google#19269]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team) to expedite investigation and resolution of this issue.

### Terraform Version & Provider Version(s)

Terraform v1.9.5
+ provider registry.terraform.io/hashicorp/google v5.43.0
+ provider registry.terraform.io/hashicorp/google-beta v5.43.0

### Affected Resource(s)

google_pubsub_subscription

### Terraform Configuration

```tf
resource "google_pubsub_subscription" "test-subscriptions" {
  for_each = toset([
    "topic.with.filter",
    "topic.without.filter"
  ])

  name    = "${each.value}-subscription-test-service"
  project = local.project_name
  topic   = google_pubsub_topic.product-backbone-topics[each.value].name

  filter = each.value == "topic.with.filter" ? "attributes.for_service=\"test-service\"" : ""

  push_config {
    push_endpoint = "${data.google_cloud_run_service.test-service.status[0].url}/push-me"

    oidc_token {
      service_account_email = google_service_account.serviceaccounts["my-pubsub-sa"].email
    }
  }
}
```

### Expected Behavior

Terraform creates the two subscriptions on the two topics, one with a filter and one without any filter.
This worked in previous versions (v5.42.0).

### Actual Behavior

Instead of creating the subscription without a filter, this now causes an error: 
```
Error: "filter" ("") doesn't match regexp "^.{1,256}$"
```

### Steps to reproduce

1. `terraform apply`

**Comment by jakobatfreiheit** [comment]
Introduced by this commit: https://github.com/hashicorp/terraform-provider-google/commit/2be944a12d8ec0d2f07d6d13567071bc2791927e

**Comment by c2thorn** [comment]
This is probably worth backporting into 5.X. Along with the resolution to https://github.com/hashicorp/terraform-provider-google/issues/15208

I'll leave this open to track that

@roaks3 fyi

**Comment by jakobatfreiheit** [comment]
@rileykarson could we get a patch release for this before the next Monday release?
This is currently blocking all the TF deployments in our project.

**Comment by c2thorn** [comment]
> @rileykarson could we get a patch release for this before the next Monday release? This is currently blocking all the TF deployments in our project.

we'll likely wait for the other issue to be resolved so we can bundle both resolutions into a single patch release.
In the meantime, are you able to [pin](https://developer.hashicorp.com/terraform/tutorials/configuration-language/versions#terraform-version-constraints) to the previous provider version to unblock yourself?

**Comment by c2thorn** [comment]
This was released as a [backport in 5.43.1](https://github.com/hashicorp/terraform-provider-google/releases/tag/v5.43.1) and is scheduled to be released in 6.1.0 tuesday
